### PR TITLE
WolmoActivityHandler converted to kotlin

### PR DESCRIPTION
**Pull request by patofernandez**

### CODE REVIEW

File: core/src/main/java/ar/com/wolox/wolmo/core/activity/WolmoActivityHandler.kt [hunk 1]
```diff
@@ -19,49 +19,40 @@
...
-
-    private ToastFactory mToastFactory;
-    private WolmoActivity mWolmoActivity;
-
-    @Inject
-    WolmoActivityHandler(ToastFactory toastFactory) {
-        mToastFactory = toastFactory;
-    }
+class WolmoActivityHandler @Inject constructor(
+        private val mToastFactory: ToastFactory
```
Comment (dylan-muszel) [inline core/src/main/java/ar/com/wolox/wolmo/core/activity/WolmoActivityHandler.kt]: Remember the `m...` convention is just for Java, not for kotlin.

File: core/src/main/java/ar/com/wolox/wolmo/core/activity/WolmoActivityHandler.kt [hunk 2]
```diff
@@ -19,49 +19,40 @@
...
+) {
+    private lateinit var wolmoActivity: WolmoActivity
 
     /**
      * Handles the custom lifecycle of Wolmo's Activity. It provides a set of callbacks to structure
      * the different aspects of the Activities initialization.
      *
-     * @param savedInstanceState a {@link Bundle} provided by Android's lifecycle.
-     * @param activity the {@link WolmoActivity} to be managed by this handler.
+     * savedInstanceState a [Bundle] provided by Android's lifecycle.
```
Comment (dylan-muszel) [inline core/src/main/java/ar/com/wolox/wolmo/core/activity/WolmoActivityHandler.kt]: Also remember [Kdoc convention](https://kotlinlang.org/docs/reference/coding-conventions.html#documentation-comments) says "incorporate the description of parameters and return values directly into the documentation comment".
I write an example about it:
```
/**
* Handles the custom lifecycle of Wolmo's Activity. It provides a set of callbacks to structure
* the different aspects of the Activities initialization. It expects the [activity] to be managed by te
* this and the [savedInstanceState] provided by Android's lifecycle.
*/

**Comment by dylan-muszel** [comment]
Also, @patofernandez, change the base branch to `kotlin-migration`.

File: core/src/main/java/ar/com/wolox/wolmo/core/activity/WolmoActivityHandler.kt (hunk 2 above)
Comment (patofernandez) [reply core/src/main/java/ar/com/wolox/wolmo/core/activity/WolmoActivityHandler.kt]: thanks! done!

File: core/src/main/java/ar/com/wolox/wolmo/core/activity/WolmoActivityHandler.kt (hunk 1 above)
Comment (patofernandez) [reply core/src/main/java/ar/com/wolox/wolmo/core/activity/WolmoActivityHandler.kt]: I had not seen it, done!

**Pull request merged**